src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs

```diff
@@ -288,5 +289,18 @@ public static string FirstCharToLowerCase(this string str)
 
             return char.ToLower(str[0]) + str.Substring(1);
         }
+
+        public static string FirstCharToUpperCase(this string str)
+        {
+            if (string.IsNullOrEmpty(str) || char.IsUpper(str[0]))
+                return str;
+
+            return char.ToUpper(str[0]) + str.Substring(1);
+        }
+
+        public static IEnumerable<string> SplitByCamelCase(this string camelCase)
```

Comment:
Shall we check `null` here?

Reply:
Actually we are doing things in explicit null now - therefore the input value can never be null.

Reply:
And we do not need to check whether this is empty here, I suppose it is reasonable that this returns empty collection if we get empty input.